walk/config.go

```diff
@@ -144,29 +146,57 @@ func (cr *Configurer) Configure(c *config.Config, rel string, f *rule.File) {
 					log.Printf("the exclusion pattern is not valid %q: %s", path.Join(rel, d.Value), err)
 					continue
 				}
-				wcCopy.excludes = append(wcCopy.excludes, path.Join(rel, d.Value))
+				wc.excludes = append(wc.excludes, path.Join(rel, d.Value))
 			case "follow":
 				if err := checkPathMatchPattern(path.Join(rel, d.Value)); err != nil {
 					log.Printf("the follow pattern is not valid %q: %s", path.Join(rel, d.Value), err)
 					continue
 				}
-				wcCopy.follow = append(wcCopy.follow, path.Join(rel, d.Value))
+				wc.follow = append(wc.follow, path.Join(rel, d.Value))
 			case "ignore":
 				if d.Value != "" {
 					log.Printf("the ignore directive does not take any arguments. Did you mean to use gazelle:exclude instead? in //%s '# gazelle:ignore %s'", f.Pkg, d.Value)
 				}
-				wcCopy.ignore = true
+				wc.ignore = true
 			}
 		}
 	}
+}
 
-	c.Exts[walkName] = wcCopy
+type ignoreFilter struct {
+	isBazelIgnored         isIgnoredFunc
+	isRepoDirectoryIgnored isIgnoredFunc
 }
 
 type isIgnoredFunc = func(string) bool
 
 var nothingIgnored isIgnoredFunc = func(string) bool { return false }
 
+func newIgnoreFilter(repoRoot string) *ignoreFilter {
```

Comment:
Optional: this comment mostly applies to pre-existing code.

`ignoreFilter` seems too indirect to me. It's not possible from reading the code of its constructor or methods to understand what it does. I think `loadBazelIgnore` and `loadRepoDirectoryIgnore` should return the data needed to fully implement `isDirectoryIgnored` and `isFileIgnored` instead of returning closures.

Reply:
Sounds good, I'll copy this comment there.